File: leetcode/easy/largest-odd-number-in-string.py
# https://leetcode.com/problems/largest-odd-number-in-string/

class Solution:
    # MINE..
    '''
    def largestOddNumber(self, num: str) -> str:
        for idx, char in enumarete(num):
            print(idx, char)
            if num[idx]%2 === 1:
                return num.substring(0)
    '''

    # ANSWER
    def largestOddNumber(self, num: str) -> str:
        for i in range(len(num) - 1, -1, -1):
            if int(num[i]) % 2 == 1:
                return num[:i+1]
        return ''

# range(start, stop, step)

File: leetcode/easy/largest-odd-number-in-string.ts
// https://leetcode.com/problems/largest-odd-number-in-string/

// MINE
function largestOddNumber(num: string): string {
  for (let i = num.length - 1; i >= 0; --i) {
    if (+num[i] % 2 === 1) return num.substring(0, i + 1)
  }
  return ''
}

// ANSWER
